crates/spirv-std/src/derivative.rs

```diff
@@ -74,25 +77,23 @@ macro_rules! deriv_fn {
     };
 }
 
+#[macro_export]
+#[doc(hidden)]
 macro_rules! deriv_impl {
     ($ty:ty) => {
-        impl Derivative for $ty {
-            deriv_fn!(ddx, OpDPdx, false);
-            deriv_fn!(ddx_fine, OpDPdxFine, true);
-            deriv_fn!(ddx_coarse, OpDPdxCoarse, true);
-            deriv_fn!(ddy, OpDPdy, false);
-            deriv_fn!(ddy_fine, OpDPdyFine, true);
-            deriv_fn!(ddy_coarse, OpDPdyCoarse, true);
-            deriv_fn!(fwidth, OpFwidth, false);
-            deriv_fn!(fwidth_fine, OpFwidthFine, true);
-            deriv_fn!(fwidth_coarse, OpFwidthCoarse, true);
+        impl $crate::derivative::Derivative for $ty {
+            $crate::deriv_fn!(ddx, OpDPdx, false);
+            $crate::deriv_fn!(ddx_fine, OpDPdxFine, true);
+            $crate::deriv_fn!(ddx_coarse, OpDPdxCoarse, true);
+            $crate::deriv_fn!(ddy, OpDPdy, false);
+            $crate::deriv_fn!(ddy_fine, OpDPdyFine, true);
+            $crate::deriv_fn!(ddy_coarse, OpDPdyCoarse, true);
+            $crate::deriv_fn!(fwidth, OpFwidth, false);
+            $crate::deriv_fn!(fwidth_fine, OpFwidthFine, true);
+            $crate::deriv_fn!(fwidth_coarse, OpFwidthCoarse, true);
         }
     };
 }
 
 // "must be a scalar or vector of floating-point type. The component width must be 32 bits."
 deriv_impl!(f32);
-deriv_impl!(glam::Vec2);
-deriv_impl!(glam::Vec3);
-deriv_impl!(glam::Vec3A);
-deriv_impl!(glam::Vec4);
```

Comment:
I would prefer to keep these in `spirv-std`, as they're not general-purpose functions that can be used by any consumer of glam.

Reply:
Sure, they are hidden from the public API, so they wouldn't have a stability guarantee, but I'll just copy a version of the macro into `glam` to add the trait impls. This will also remove the changes around macros that were needed to handle correctly calling `derive_impl!` from `glam`.